File: src/main/java/com/bajdcc/LALR1/semantic/tracker/ErrorRecord.java
package com.bajdcc.LALR1.semantic.tracker;

import java.util.ArrayList;

/**
 * 错误记录器链表
 *
 * @author bajdcc
 */
public class ErrorRecord {

	/**
	 * 错误集
	 */
	public ArrayList<TrackerError> arrErrors = new ArrayList<>();

	/**
	 * 前向指针
	 */
	public ErrorRecord prev = null;

	public ErrorRecord(ErrorRecord prev) {
		this.prev = prev;
	}
}

File: src/main/kotlin/com/bajdcc/LALR1/semantic/tracker/ErrorRecord.kt
package com.bajdcc.LALR1.semantic.tracker

/**
 * 错误记录器链表
 * [prev] 前向指针
 * [errors] 错误集
 * @author bajdcc
 */
data class ErrorRecord(var prev: ErrorRecord?,
                       var errors: MutableList<TrackerError> = mutableListOf<TrackerError>())